project-base/storefront/hooks/auth/useLogout.tsx

```diff
@@ -0,0 +1,35 @@
+import { useLogoutMutation } from 'graphql/requests/auth/mutations/LogoutMutation.generated';
+import { removeTokensFromCookies } from 'helpers/auth/removeTokensFromCookies';
+import { dispatchBroadcastChannel } from 'hooks/useBroadcastChannel';
+import { useRouter } from 'next/router';
+import { usePersistStore } from 'store/usePersistStore';
+import { useSessionStore } from 'store/useSessionStore';
+
+type LogoutHandler = () => Promise<void>;
+
+export const useLogout = () => {
+    const [, logoutMutation] = useLogoutMutation();
+
+    const updateAuthLoadingState = usePersistStore((store) => store.updateAuthLoadingState);
+    const updatePageLoadingState = useSessionStore((s) => s.updatePageLoadingState);
+    const updateProductListUuids = usePersistStore((s) => s.updateProductListUuids);
+
+    const router = useRouter();
+
+    const logout: LogoutHandler = async () => {
```

Comment:
Is it necessary to have type here? If yes, then I suggest remove separate type `LogoutHandler` and define return value type here.

Reply:
Yeah, probably not necessary. I removed it.